packages/accounts-base/accounts_server.js

```diff
@@ -1135,18 +1135,14 @@ Ap._expirePasswordResetTokens = function (oldestValidDate, userId) {
   var userFilter = userId ? {_id: userId} : {};
 
   this.users.update(_.extend(userFilter, {
+    "services.password.reset.reason": reason,
```

Comment:
This change will now not expire tokens made before we introduced this new field.

Maybe it would be better if we would simply have one query, which would be something like: remove all tokens if (reason is enroll and older than X) OR (reason is not enroll (including no reason) and older than Y).